src/apphosting/backend.ts

```diff
@@ -474,3 +474,33 @@ export async function getBackendForAmbiguousLocation(
   });
   return backendsByLocation.get(location)!;
 }
+
+/**
+ * Fetches a backend from the server. If there are multiple backends with the name, it will throw an error
+ * telling the user that there are other backends with the same name that need to be deleted.
+ */
+export async function getBackend(
+  projectId: string,
+  backendId: string,
+): Promise<apphosting.Backend> {
+  let { unreachable, backends } = await apphosting.listBackends(projectId, "-");
+  backends = backends.filter(
+    (backend) => apphosting.parseBackendName(backend.name).id === backendId,
+  );
+  if (backends.length > 1) {
+    throw new FirebaseError(
+      `You have multiple backends with the same ${backendId} ID. This is not allowed until we can support more locations. ` +
```

Comment:
since we already have the backends, think it would be nice to list out locations where the conflict is arising.